network-config-analyzer/K8sService.py

```diff
@@ -31,7 +31,8 @@ def __init__(self, port, target_port, protocol, name=''):
 
     def __init__(self, name, namespace_name):
         self.name = name
-        self.namespace = K8sNamespace(namespace_name)
+        self.namespace_name = namespace_name
+        self.namespace = None
```

Comment:
Added a comment

Reply:
I still don't get it.
Why is `Service` so special? Why do we need this late binding? Why can't `K8sServiceYamlParser` set `self.namespace` directly when it constructs the object?

Reply:
Service makes late binding of target pods, because serviced might have been parsed before pods. Therefore, K8sServiceYamlParser does not access PeerContainer (which is also keeps all namespaces).